### ISSUE TITLE
Using Ctrl-Q on Firefox closes the browser

### ISSUE THREAD

**Issue by madx** [jupyterlab/jupyterlab#4326]
Hi

The current keyboard shortcut for closing tabs is Ctrl-Q, which happens to also be the shortcut used by Firefox to close the browser.

I found out that I can configure it using the Settings Editor, but it's not very convenient for everyone. A bunch of related questions:

- Do you think there is a way to prevent the default action?
- Do you plan on adding a simpler keyboard shortcuts editor?
- What about detecting the browser and using a different mapping?

**Comment by ian-r-rose** [comment]
Hi @madx,

I agree, that behavior is annoying to me as well. So far as I know, some default actions can be prevented, but some are reserved by the browser, so we can't control for all of them. There are plans to add a simpler keyboard shortcut editor, but it is a significant undertaking, so it is not at the top of the priorities for the moment.

Detecting the browser sounds like an intriguing possibility. This is also related to changing the keyboard shortcuts in response to different keyboard layouts/internationalization. The last time we discussed that, we concluded that having a robust set of keyboard shortcuts for different browsers/layouts was a combinatorial nightmare, and that it was better to make them easily configurable. So at the moment, I would say we are halfway there (they are configurable, but not easily so!).
